### PR TITLE
Fix sm_SUITE

### PR DESCRIPTION
**Pull request by arcusfelis**
Proposed changes include:

* **Wait for resume_session state in resend_unacked_on_reconnection** - we basically could fail otherwise by suspending c2s before a receiver process gets down.
* Log messages if wait_for_queue_length fails
* A separate stop_client_and_wait_for_termination function for easy reading
*   Fix loglevel in sm_SUITE

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93075) / small_tests / b4b911d843e4af59e602453752c38086989b7e8a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3436/93075/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3436/93075/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-07_18.11.43/lib.mongooseim.logs/run.2021-12-07_18.11.43/suite.log.html)

File: big_tests/tests/sm_SUITE.erl [hunk 1]
```diff
@@ -1474,10 +1474,12 @@ given_fresh_user_with_spec(Spec) ->
     {User#client{jid  = JID}, Spec}.
 
 wait_for_queue_length(Pid, Length) ->
+    DebugFun = fun() -> rpc(mim(), erlang, process_info, [Pid, messages]) end,
+    Opts = #{debug_fun => DebugFun},
     mongoose_helper:wait_until(
       fun() ->
               rpc(mim(), erlang, process_info, [Pid, message_queue_len])
-      end, {message_queue_len, Length}).
+      end, {message_queue_len, Length}, Opts).
```
Comment (NelsonVides) [inline big_tests/tests/sm_SUITE.erl:1482]: Not sure I like this one, is doing two rpc calls, and they're very related. You might as well get the messages, and check if the list is empty. If it is, then condition is met, if it is not, then store that list in the error log. I'd find that a better practice, cleaner, what do you think?

Comment (arcusfelis) [reply big_tests/tests/sm_SUITE.erl:1482]: why empty, lol? The usage is 2 any messages. Good luck debuging if it is three.

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93587) / small_tests / 41a3d5611caf8cc81ebf5b59feefa1cd09423004
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3436/93587/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3436/93587/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-09_11.24.23/lib.mongooseim.logs/run.2021-12-09_11.24.23/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93605) / small_tests / 9f414a0b7451fb2482489192ff94d9f8bfb008a7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3436/93605/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3436/93605/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-09_11.49.24/lib.mongooseim.logs/run.2021-12-09_11.49.24/suite.log.html)

File: big_tests/tests/mongoose_helper.erl [hunk 2]
```diff
@@ -382,6 +382,13 @@ do_wait_until(Fun, ExpectedValue, Opts) ->
             wait_and_continue(Fun, ExpectedValue, {Error, Reason}, Opts)
     end.
 
+simplify_history([H|[H|_]=T], Times) ->
+    simplify_history(T, Times + 1);
+simplify_history([H|T], Times) ->
+    [{times, Times, H}|simplify_history(T, 1)];
+simplify_history([], 1) ->
+    [].
+
```
Comment (arcusfelis) [inline big_tests/tests/mongoose_helper.erl:391]: that prints:

```
{{{wait_for_queue_length,5},
  ok,
  [{times,1,
       {messages,2,
           [{'$gen_all_state_event',dummy_event},
            {'$gen_sync_event',
                {<8430.3222.11>,#Ref<8430.39755002.4269539331.51912>},
                resume}]}},
   {times,49,
       {messages,3,
           [{'$gen_all_state_event',dummy_event},
            {'$gen_sync_event',
                {<8430.3222.11>,#Ref<8430.39755002.4269539331.51912>},
                resume},
            {route,
                {jid,
                    <<"bOb_messages_are_properly_flushed_during_resumption_p1_fsm_old_1209120542.110">>,
                    <<"localhost">>,<<"res1">>,
                    <<"bob_messages_are_properly_flushed_during_resumption_p1_fsm_old_1209120542.110">>,
                    <<"localhost">>,<<"res1">>},
                {jid,
                    <<"alicE_messages_are_properly_flushed_during_resumption_p1_fsm_old_1209120542.103">>,
                    <<"localhost">>,<<"escalus-default-resource">>,
                    <<"alice_messages_are_properly_flushed_during_resumption_p1_fsm_old_1209120542.103">>,
                    <<"localhost">>,<<"escalus-default-resource">>},
                #{host_type => <<"localhost">>,lserver => <<"localhost">>,
                  mongoose_acc => true,
                  non_strippable => [{c2s,origin_jid},{c2s,origin_sid}],
                  origin_location =>
                      #{file =>
                            "/Users/mikhailuvarov/erlang/esl/MongooseIM/src/ejabberd_c2s.erl",
                        line => 3304,
                        mfa => {ejabberd_c2s,element_to_origin_accum,2}},
                  origin_pid => <8430.3167.11>,
                  origin_stanza => ...
```

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/93640) / small_tests / 9de0857c8fc2e4dfad6859108ffc945b489c833d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3436/93640/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3436/93640/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-09_12.10.41/lib.mongooseim.logs/run.2021-12-09_12.10.42/suite.log.html)

**Review by NelsonVides — APPROVED** [review]
Yeah perfect 👌🏽

**Pull request merged**